benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py

```diff
@@ -51,44 +53,68 @@
 
 
 class BenchmarkBase(ABC):
+    # measure total number of instruction spent in _work.
     _instruction_count = False
 
+    # measure total number of instruction spent in convert_frame.compile_inner
+    # TODO is there other parts we need to add ?
+    _compile_time_instruction_count = False
+
     def enable_instruction_count(self):
         self._instruction_count = True
         return self
 
+    def enable_compile_time_instruction_count(self):
+        self._compile_time_instruction_count = True
+        return self
+
     def name(self):
         return ""
 
     def description(self):
         return ""
 
     @abstractmethod
-    def prepare(self):
+    def _prepare(self):
         pass
 
     @abstractmethod
-    def work(self):
+    def _work(self):
         pass
 
-    def prepare_once(self):  # noqa: B027
+    def _prepare_once(self):  # noqa: B027
         pass
 
-    def count_instructions(self):
+    def _count_instructions(self):
         print(f"collecting instruction count for {self.name()}")
-        self.prepare_once()
-
         results = []
         for i in range(10):
-            self.prepare()
+            self._prepare()
             id = i_counter.start()
-            self.work()
+            self._work()
             count = i_counter.end(id)
             print(f"instruction count for iteration {i} is {count}")
             if i != 0:
                 results.append(count)
         return min(results)
 
+    def _count_compile_time_instructions(self):
+        print(f"collecting compile time instruction count for {self.name()}")
+        config.record_compile_time_instruction_count = True
+
+        results = []
+        for i in range(10):
+            self._prepare()
+            CompileTimeInstructionCounter.clear()
+            self._work()
+            count = CompileTimeInstructionCounter.value()
+            print(f"compile time instruction count for iteration {i} is {count}")
+            if i != 0:
```

Comment:
Why skip 0? Warmup? Since we're returning `min` warmup shouldn't matter... Please add a comment.

Reply:
yeh it does not really matter we can remove it